### PR TITLE
ceph: Enable Ceph Pacific as a supported version

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
With the Ceph Pacific release coming this week we add support in Rook for Pacific with the Rook v1.6 release coming soon. The integration tests will now run across nautilus, octopus, and pacific to cover all supported Ceph versions. The default examples still specify Octopus until there is more bake time for Pacific.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Comment by travisn** [comment]
Retrieving bucket stats is failing in the integration tests for pacific. 
@thotz @alimaredia Did something change in Pacific with the `radosgw-admin bucket stats` command that Rook needs a different command, or is this a Pacific bug?

Listing the buckets works fine:
```
[root@rook-ceph-tools-78cdfd976c-v8fcn /]# radosgw-admin bucket list         
[
    "rook-ceph-bucket-checker-ce00839d-5235-4c5c-a08b-7577d71a304f"
]
```

Getting the bucket stats fails:
```
[root@rook-ceph-tools-78cdfd976c-v8fcn /]# radosgw-admin bucket stats
[]
2021-03-30T23:22:56.968+0000 7f5c45c731c0  0 ERROR: could not decode buffer info, caught buffer::error
2021-03-30T23:22:56.968+0000 7f5c45c731c0 -1 ERROR: do_read_bucket_instance_info failed: -5
```

This is using the tag `ceph/daemon-base:latest-pacific-devel` which has the version: `ceph version 16.1.0-1323-g7e7e1f4e (7e7e1f4ed3cc94c90a0e806a8c52e1798323bcc0) pacific (rc)`.

File: tests/scripts/validate_cluster.sh [hunk 1]
```diff
@@ -69,12 +69,7 @@ function test_demo_mds {
   # so we first check if the pools exit, from that we assume that
   # the process will start. We stop waiting after 10 seconds.
   # shellcheck disable=SC2046
-  return $(wait_for_daemon "$EXEC_COMMAND osd dump | grep -sq cephfs && $EXEC_COMMAND -s | grep -sq 'up:active'")
-}
-
-function test_demo_rgw {
```
[the remark this answers is not in the document]
Comment (travisn) [reply tests/scripts/validate_cluster.sh:75]: this was a duplicate function defined for `test_demo_rgw`, so the duplicate is just removed

[the remark this answers is not in the document]
Comment (leseb) [reply tests/scripts/validate_cluster.sh:75]: Ah right. Please make it a separate commit then.

**Review by leseb — CHANGES_REQUESTED** [review]

**Comment by travisn** [comment]
The tests are all passing in the github actions, and now the block and file tests are all failing in Jenkins. The issue seems to be the kernel version required by Pacific.

The [rbd plugin](https://jenkins.rook.io/job/rook/job/rook/job/PR-7510/12/artifact/_output/tests/TestCephSmokeSuite_aws_1.15.x_smoke-ns-system_csi-rbdplugin-xpjv5_1617770207.log) log shows the following error when mounting the volume:
```
I0407 04:17:49.708470   22454 utils.go:132] ID: 15 GRPC call: /csi.v1.Node/NodeGetCapabilities
I0407 04:17:49.708709   22454 utils.go:133] ID: 15 GRPC request: {}
I0407 04:17:49.708851   22454 utils.go:138] ID: 15 GRPC response: {"capabilities":[{"Type":{"Rpc":{"type":1}}},{"Type":{"Rpc":{"type":2}}},{"Type":{"Rpc":{"type":3}}}]}
I0407 04:17:49.813407   22454 utils.go:132] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 GRPC call: /csi.v1.Node/NodeStageVolume
I0407 04:17:49.813557   22454 utils.go:133] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 GRPC request: {"secrets":"***stripped***","staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-4f961e2c-6288-4395-88d5-ce6f7674eb76/globalmount","volume_capability":{"AccessType":{"Mount":{"fs_type":"ext4"}},"access_mode":{"mode":1}},"volume_context":{"clusterID":"smoke-ns","csi.storage.k8s.io/pv/name":"pvc-4f961e2c-6288-4395-88d5-ce6f7674eb76","csi.storage.k8s.io/pvc/name":"block-pv-claim","csi.storage.k8s.io/pvc/namespace":"default","imageFeatures":"layering","imageName":"csi-vol-1ed7a671-9758-11eb-883b-56d8561839b9","journalPool":"replicapool","pool":"replicapool","radosNamespace":"","storage.kubernetes.io/csiProvisionerIdentity":"1617768944751-8081-smoke-ns-system.rbd.csi.ceph.com"},"volume_id":"0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9"}
I0407 04:17:49.813745   22454 rbd_util.go:808] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 setting disableInUseChecks on rbd volume to: false
I0407 04:17:49.814649   22454 omap.go:84] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 got omap values: (pool="replicapool", namespace="", name="csi.volume.1ed7a671-9758-11eb-883b-56d8561839b9"): map[csi.imageid:385c316e278b csi.imagename:csi-vol-1ed7a671-9758-11eb-883b-56d8561839b9 csi.volname:pvc-4f961e2c-6288-4395-88d5-ce6f7674eb76]
E0407 04:17:49.814887   22454 util.go:232] kernel 4.15.0-1027-aws does not support required features
I0407 04:17:49.855450   22454 cephcmds.go:59] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 command succeeded: rbd [device list --format=json --device-type krbd]
I0407 04:17:49.907956   22454 rbd_attach.go:215] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 rbd: map mon 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789
I0407 04:17:49.951301   22454 cephcmds.go:53] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 an error (exit status 95) occurred while running rbd args: [--id csi-rbd-node -m 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789 --keyfile=***stripped*** map replicapool/csi-vol-1ed7a671-9758-11eb-883b-56d8561839b9 --device-type krbd]
W0407 04:17:49.951322   22454 rbd_attach.go:245] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 rbd: map error an error (exit status 95) occurred while running rbd args: [--id csi-rbd-node -m 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789 --keyfile=***stripped*** map replicapool/csi-vol-1ed7a671-9758-11eb-883b-56d8561839b9 --device-type krbd], rbd output: rbd: sysfs write failed
rbd: map failed: (95) Operation not supported
E0407 04:17:49.951449   22454 utils.go:136] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 GRPC error: rpc error: code = Internal desc = rbd: map failed with error an error (exit status 95) occurred while running rbd args: [--id csi-rbd-node -m 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789 --keyfile=***stripped*** map replicapool/csi-vol-1ed7a671-9758-11eb-883b-56d8561839b9 --device-type krbd], rbd error output: rbd: sysfs write failed
rbd: map failed: (95) Operation not supported
```

The [cephfs plugin](https://jenkins.rook.io/job/rook/job/rook/job/PR-7510/12/artifact/_output/tests/TestCephSmokeSuite_aws_1.15.x_smoke-ns-system_csi-cephfsplugin-d699h_1617770206.log) log shows the following error when mounting the volume:
```
I0407 04:29:58.254094       1 utils.go:132] ID: 33 GRPC call: /csi.v1.Node/NodeGetCapabilities
I0407 04:29:58.254149       1 utils.go:133] ID: 33 GRPC request: {}
I0407 04:29:58.254249       1 utils.go:138] ID: 33 GRPC response: {"capabilities":[{"Type":{"Rpc":{"type":1}}},{"Type":{"Rpc":{"type":2}}}]}
I0407 04:29:58.377445       1 utils.go:132] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 GRPC call: /csi.v1.Node/NodeStageVolume
I0407 04:29:58.377567       1 utils.go:133] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 GRPC request: {"secrets":"***stripped***","staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-5acef563-3bd0-4862-8c21-7ece4f289f24/globalmount","volume_capability":{"AccessType":{"Mount":{}},"access_mode":{"mode":1}},"volume_context":{"clusterID":"smoke-ns","fsName":"smoke-test-fs","pool":"smoke-test-fs-data0","storage.kubernetes.io/csiProvisionerIdentity":"1617768944598-8081-smoke-ns-system.cephfs.csi.ceph.com","subvolumeName":"csi-vol-90b3146c-9759-11eb-b3ec-8e4f20c6c758"},"volume_id":"0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758"}
I0407 04:29:58.382709       1 omap.go:84] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 got omap values: (pool="smoke-test-fs-metadata", namespace="csi", name="csi.volume.90b3146c-9759-11eb-b3ec-8e4f20c6c758"): map[csi.imagename:csi-vol-90b3146c-9759-11eb-b3ec-8e4f20c6c758 csi.volname:pvc-5acef563-3bd0-4862-8c21-7ece4f289f24]
I0407 04:29:58.396983       1 volumemounter.go:133] requested mounter: , chosen mounter: kernel
I0407 04:29:58.397014       1 nodeserver.go:153] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 cephfs: mounting volume 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 with Ceph kernel client
I0407 04:29:58.398554       1 cephcmds.go:59] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 command succeeded: modprobe [ceph]
I0407 04:29:58.416733       1 cephcmds.go:53] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 an error (exit status 32) occurred while running mount args: [-t ceph 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789:/volumes/csi/csi-vol-90b3146c-9759-11eb-b3ec-8e4f20c6c758/8941a78e-6168-459c-9537-c621207fcc2c /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-5acef563-3bd0-4862-8c21-7ece4f289f24/globalmount -o name=csi-cephfs-node,secretfile=/tmp/csi/keys/keyfile-563822658,mds_namespace=smoke-test-fs,_netdev]
E0407 04:29:58.416762       1 nodeserver.go:176] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 failed to mount volume 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758: an error (exit status 32) occurred while running mount args: [-t ceph 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789:/volumes/csi/csi-vol-90b3146c-9759-11eb-b3ec-8e4f20c6c758/8941a78e-6168-459c-9537-c621207fcc2c /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-5acef563-3bd0-4862-8c21-7ece4f289f24/globalmount -o name=csi-cephfs-node,secretfile=/tmp/csi/keys/keyfile-563822658,mds_namespace=smoke-test-fs,_netdev] Check dmesg logs if required.
E0407 04:29:58.416793       1 utils.go:136] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 GRPC error: rpc error: code = Internal desc = an error (exit status 32) occurred while running mount args: [-t ceph 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789:/volumes/csi/csi-vol-90b3146c-9759-11eb-b3ec-8e4f20c6c758/8941a78e-6168-459c-9537-c621207fcc2c /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-5acef563-3bd0-4862-8c21-7ece4f289f24/globalmount -o name=csi-cephfs-node,secretfile=/tmp/csi/keys/keyfile-563822658,mds_namespace=smoke-test-fs,_netdev]
```

**Comment by travisn** [comment]
In one of the operator logs in the artifacts I see this crash that caused an operator restart. It could explain flakiness in other tests.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1dcbd5f]

goroutine 1276 [running]:
github.com/rook/rook/pkg/operator/ceph/pool.(*mirrorChecker).checkMirroringHealth(0xc0019adda0, 0xb, 0xc00113f950)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/pool/health.go:115 +0xdf
github.com/rook/rook/pkg/operator/ceph/pool.(*mirrorChecker).checkMirroring(0xc0019adda0, 0xc001250f60)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/pool/health.go:82 +0x18f
created by github.com/rook/rook/pkg/operator/ceph/pool.(*ReconcileCephBlockPool).reconcile
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/pool/controller.go:280 +0xc08
```

**Comment by Madhu-1** [comment]
> The tests are all passing in the github actions, and now the block and file tests are all failing in Jenkins. The issue seems to be the kernel version required by Pacific.
> 
> The [rbd plugin](https://jenkins.rook.io/job/rook/job/rook/job/PR-7510/12/artifact/_output/tests/TestCephSmokeSuite_aws_1.15.x_smoke-ns-system_csi-rbdplugin-xpjv5_1617770207.log) log shows the following error when mounting the volume:
> 
> ```
> I0407 04:17:49.708470   22454 utils.go:132] ID: 15 GRPC call: /csi.v1.Node/NodeGetCapabilities
> I0407 04:17:49.708709   22454 utils.go:133] ID: 15 GRPC request: {}
> I0407 04:17:49.708851   22454 utils.go:138] ID: 15 GRPC response: {"capabilities":[{"Type":{"Rpc":{"type":1}}},{"Type":{"Rpc":{"type":2}}},{"Type":{"Rpc":{"type":3}}}]}
> I0407 04:17:49.813407   22454 utils.go:132] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 GRPC call: /csi.v1.Node/NodeStageVolume
> I0407 04:17:49.813557   22454 utils.go:133] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 GRPC request: {"secrets":"***stripped***","staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-4f961e2c-6288-4395-88d5-ce6f7674eb76/globalmount","volume_capability":{"AccessType":{"Mount":{"fs_type":"ext4"}},"access_mode":{"mode":1}},"volume_context":{"clusterID":"smoke-ns","csi.storage.k8s.io/pv/name":"pvc-4f961e2c-6288-4395-88d5-ce6f7674eb76","csi.storage.k8s.io/pvc/name":"block-pv-claim","csi.storage.k8s.io/pvc/namespace":"default","imageFeatures":"layering","imageName":"csi-vol-1ed7a671-9758-11eb-883b-56d8561839b9","journalPool":"replicapool","pool":"replicapool","radosNamespace":"","storage.kubernetes.io/csiProvisionerIdentity":"1617768944751-8081-smoke-ns-system.rbd.csi.ceph.com"},"volume_id":"0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9"}
> I0407 04:17:49.813745   22454 rbd_util.go:808] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 setting disableInUseChecks on rbd volume to: false
> I0407 04:17:49.814649   22454 omap.go:84] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 got omap values: (pool="replicapool", namespace="", name="csi.volume.1ed7a671-9758-11eb-883b-56d8561839b9"): map[csi.imageid:385c316e278b csi.imagename:csi-vol-1ed7a671-9758-11eb-883b-56d8561839b9 csi.volname:pvc-4f961e2c-6288-4395-88d5-ce6f7674eb76]
> E0407 04:17:49.814887   22454 util.go:232] kernel 4.15.0-1027-aws does not support required features
> I0407 04:17:49.855450   22454 cephcmds.go:59] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 command succeeded: rbd [device list --format=json --device-type krbd]
> I0407 04:17:49.907956   22454 rbd_attach.go:215] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 rbd: map mon 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789
> I0407 04:17:49.951301   22454 cephcmds.go:53] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 an error (exit status 95) occurred while running rbd args: [--id csi-rbd-node -m 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789 --keyfile=***stripped*** map replicapool/csi-vol-1ed7a671-9758-11eb-883b-56d8561839b9 --device-type krbd]
> W0407 04:17:49.951322   22454 rbd_attach.go:245] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 rbd: map error an error (exit status 95) occurred while running rbd args: [--id csi-rbd-node -m 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789 --keyfile=***stripped*** map replicapool/csi-vol-1ed7a671-9758-11eb-883b-56d8561839b9 --device-type krbd], rbd output: rbd: sysfs write failed
> rbd: map failed: (95) Operation not supported
> E0407 04:17:49.951449   22454 utils.go:136] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 GRPC error: rpc error: code = Internal desc = rbd: map failed with error an error (exit status 95) occurred while running rbd args: [--id csi-rbd-node -m 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789 --keyfile=***stripped*** map replicapool/csi-vol-1ed7a671-9758-11eb-883b-56d8561839b9 --device-type krbd], rbd error output: rbd: sysfs write failed
> rbd: map failed: (95) Operation not supported
> ```
> 
> The [cephfs plugin](https://jenkins.rook.io/job/rook/job/rook/job/PR-7510/12/artifact/_output/tests/TestCephSmokeSuite_aws_1.15.x_smoke-ns-system_csi-cephfsplugin-d699h_1617770206.log) log shows the following error when mounting the volume:
> 
> ```
> I0407 04:29:58.254094       1 utils.go:132] ID: 33 GRPC call: /csi.v1.Node/NodeGetCapabilities
> I0407 04:29:58.254149       1 utils.go:133] ID: 33 GRPC request: {}
> I0407 04:29:58.254249       1 utils.go:138] ID: 33 GRPC response: {"capabilities":[{"Type":{"Rpc":{"type":1}}},{"Type":{"Rpc":{"type":2}}}]}
> I0407 04:29:58.377445       1 utils.go:132] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 GRPC call: /csi.v1.Node/NodeStageVolume
> I0407 04:29:58.377567       1 utils.go:133] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 GRPC request: {"secrets":"***stripped***","staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-5acef563-3bd0-4862-8c21-7ece4f289f24/globalmount","volume_capability":{"AccessType":{"Mount":{}},"access_mode":{"mode":1}},"volume_context":{"clusterID":"smoke-ns","fsName":"smoke-test-fs","pool":"smoke-test-fs-data0","storage.kubernetes.io/csiProvisionerIdentity":"1617768944598-8081-smoke-ns-system.cephfs.csi.ceph.com","subvolumeName":"csi-vol-90b3146c-9759-11eb-b3ec-8e4f20c6c758"},"volume_id":"0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758"}
> I0407 04:29:58.382709       1 omap.go:84] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 got omap values: (pool="smoke-test-fs-metadata", namespace="csi", name="csi.volume.90b3146c-9759-11eb-b3ec-8e4f20c6c758"): map[csi.imagename:csi-vol-90b3146c-9759-11eb-b3ec-8e4f20c6c758 csi.volname:pvc-5acef563-3bd0-4862-8c21-7ece4f289f24]
> I0407 04:29:58.396983       1 volumemounter.go:133] requested mounter: , chosen mounter: kernel
> I0407 04:29:58.397014       1 nodeserver.go:153] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 cephfs: mounting volume 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 with Ceph kernel client
> I0407 04:29:58.398554       1 cephcmds.go:59] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 command succeeded: modprobe [ceph]
> I0407 04:29:58.416733       1 cephcmds.go:53] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 an error (exit status 32) occurred while running mount args: [-t ceph 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789:/volumes/csi/csi-vol-90b3146c-9759-11eb-b3ec-8e4f20c6c758/8941a78e-6168-459c-9537-c621207fcc2c /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-5acef563-3bd0-4862-8c21-7ece4f289f24/globalmount -o name=csi-cephfs-node,secretfile=/tmp/csi/keys/keyfile-563822658,mds_namespace=smoke-test-fs,_netdev]
> E0407 04:29:58.416762       1 nodeserver.go:176] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 failed to mount volume 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758: an error (exit status 32) occurred while running mount args: [-t ceph 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789:/volumes/csi/csi-vol-90b3146c-9759-11eb-b3ec-8e4f20c6c758/8941a78e-6168-459c-9537-c621207fcc2c /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-5acef563-3bd0-4862-8c21-7ece4f289f24/globalmount -o name=csi-cephfs-node,secretfile=/tmp/csi/keys/keyfile-563822658,mds_namespace=smoke-test-fs,_netdev] Check dmesg logs if required.
> E0407 04:29:58.416793       1 utils.go:136] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 GRPC error: rpc error: code = Internal desc = an error (exit status 32) occurred while running mount args: [-t ceph 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789:/volumes/csi/csi-vol-90b3146c-9759-11eb-b3ec-8e4f20c6c758/8941a78e-6168-459c-9537-c621207fcc2c /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-5acef563-3bd0-4862-8c21-7ece4f289f24/globalmount -o name=csi-cephfs-node,secretfile=/tmp/csi/keys/keyfile-563822658,mds_namespace=smoke-test-fs,_netdev]
> ```

@Rakshith-R can you please check why CSI mounts are failing for the pacific release?

File: tests/integration/ceph_upgrade_test.go [hunk 2]
```diff
@@ -202,6 +202,17 @@ func (s *UpgradeSuite) TestUpgradeToMaster() {
 	newFile = "post-octopus-upgrade-file"
 	s.verifyFilesAfterUpgrade(filesystemName, newFile, message, rbdFilesToRead, cephfsFilesToRead)
 	logger.Infof("Verified upgrade from nautilus to octopus")
+
+	//
+	// Upgrade from octopus to pacific
+	//
+	logger.Infof("*** UPGRADING CEPH FROM Octopus TO Pacific ***")
```
Comment (leseb) [inline tests/integration/ceph_upgrade_test.go:209]:
```suggestion
	logger.Infof("*** UPGRADING CEPH FROM OCTOPUS TO PACIFIC ***")
```

**Comment by Rakshith-R** [comment]
> > The tests are all passing in the github actions, and now the block and file tests are all failing in Jenkins. The issue seems to be the kernel version required by Pacific.
> > The [rbd plugin](https://jenkins.rook.io/job/rook/job/rook/job/PR-7510/12/artifact/_output/tests/TestCephSmokeSuite_aws_1.15.x_smoke-ns-system_csi-rbdplugin-xpjv5_1617770207.log) log shows the following error when mounting the volume:
> > ```
> > I0407 04:17:49.907956   22454 rbd_attach.go:215] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 rbd: map mon 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789
> > I0407 04:17:49.951301   22454 cephcmds.go:53] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 an error (exit status 95) occurred while running rbd args: [--id csi-rbd-node -m 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789 --keyfile=***stripped*** map replicapool/csi-vol-1ed7a671-9758-11eb-883b-56d8561839b9 --device-type krbd]
> > W0407 04:17:49.951322   22454 rbd_attach.go:245] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 rbd: map error an error (exit status 95) occurred while running rbd args: [--id csi-rbd-node -m 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789 --keyfile=***stripped*** map replicapool/csi-vol-1ed7a671-9758-11eb-883b-56d8561839b9 --device-type krbd], rbd output: rbd: sysfs write failed
> > rbd: map failed: (95) Operation not supported
> > E0407 04:17:49.951449   22454 utils.go:136] ID: 16 Req-ID: 0001-0008-smoke-ns-0000000000000002-1ed7a671-9758-11eb-883b-56d8561839b9 GRPC error: rpc error: code = Internal desc = rbd: map failed with error an error (exit status 95) occurred while running rbd args: [--id csi-rbd-node -m 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789 --keyfile=***stripped*** map replicapool/csi-vol-1ed7a671-9758-11eb-883b-56d8561839b9 --device-type krbd], rbd error output: rbd: sysfs write failed
> > rbd: map failed: (95) Operation not supported
> > ```
> > 
> > 
> > The [cephfs plugin](https://jenkins.rook.io/job/rook/job/rook/job/PR-7510/12/artifact/_output/tests/TestCephSmokeSuite_aws_1.15.x_smoke-ns-system_csi-cephfsplugin-d699h_1617770206.log) log shows the following error when mounting the volume:
> > ```I0407 04:29:58.416733       1 cephcmds.go:53] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 an error (exit status 32) occurred while running mount args: [-t ceph 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789:/volumes/csi/csi-vol-90b3146c-9759-11eb-b3ec-8e4f20c6c758/8941a78e-6168-459c-9537-c621207fcc2c /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-5acef563-3bd0-4862-8c21-7ece4f289f24/globalmount -o name=csi-cephfs-node,secretfile=/tmp/csi/keys/keyfile-563822658,mds_namespace=smoke-test-fs,_netdev]
> > E0407 04:29:58.416762       1 nodeserver.go:176] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 failed to mount volume 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758: an error (exit status 32) occurred while running mount args: [-t ceph 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789:/volumes/csi/csi-vol-90b3146c-9759-11eb-b3ec-8e4f20c6c758/8941a78e-6168-459c-9537-c621207fcc2c /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-5acef563-3bd0-4862-8c21-7ece4f289f24/globalmount -o name=csi-cephfs-node,secretfile=/tmp/csi/keys/keyfile-563822658,mds_namespace=smoke-test-fs,_netdev] Check dmesg logs if required.
> > E0407 04:29:58.416793       1 utils.go:136] ID: 34 Req-ID: 0001-0008-smoke-ns-0000000000000001-90b3146c-9759-11eb-b3ec-8e4f20c6c758 GRPC error: rpc error: code = Internal desc = an error (exit status 32) occurred while running mount args: [-t ceph 10.98.181.117:6789,10.101.119.31:6789,10.103.245.68:6789:/volumes/csi/csi-vol-90b3146c-9759-11eb-b3ec-8e4f20c6c758/8941a78e-6168-459c-9537-c621207fcc2c /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-5acef563-3bd0-4862-8c21-7ece4f289f24/globalmount -o name=csi-cephfs-node,secretfile=/tmp/csi/keys/keyfile-563822658,mds_namespace=smoke-test-fs,_netdev]
> > ```
> 
> @Rakshith-R can you please check why CSI mounts are failing for the pacific release?

From the logs, CSI driver does not seem to be the reason. rbd map and kernel mount cmds are failing. 
k8s v1.15, 1.16 & 1.20 all have log the exact same error and have the same kernel version 
`> > E0407 04:17:49.814887   22454 util.go:232] kernel 4.15.0-1027-aws does not support required features`

Since Local testing and github workflows run perfectly, the issue does seem to be with the kernel version required with ceph pacific.

**Review by leseb — APPROVED** [review]

**Pull request merged**